pkg/blob/controllerserver.go

```diff
@@ -367,9 +367,15 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		mc.ObserveOperationWithResult(isOperationSucceeded, VolumeID, volumeID)
 	}()
 
+	secrets := req.GetSecrets()
+
+	volumeContentSourceSASToken, volumeContentSourcePath, err := d.validateVolumeContentSource(ctx, req, storageEndpointSuffix, secrets, secretName, secretNamespace, getLatestAccountKey)
```

Comment:
I think we don't need to add a function `validateVolumeContentSource` here. We can  move it to `copyBlobContainer` and if `srcAccountName != dstAccountName`, then generate `srcSASToken`, otherwise we just need to generate sas token once.

Reply:
The intent was to confirm we have have access to the volume source before provisioning a volume we can't populate. I could add a check to skip the sas generation for the *destination* if it matches the source. The added function was just to isolate the changes and make it easier to test, I can fold it back in if you feel strongly about it.

Reply:
@umagnus on review, the generated sas token is cached. there is no need for an additional optimization.